data/artifact/v0/config/tasks.json

```diff
@@ -1029,36 +1096,338 @@
+          "instillUIOrder": 2,
+          "instillAcceptFormats": [
+            "string"
+          ],
+          "instillUpstreamTypes": [
+            "value",
+            "reference"
+          ],
+          "title": "Role",
+          "type": "string"
+        },
+        "message-type": {
+          "description": "The message type of the chat history you want to retrieve. The default is all with blank setting. Now, we only support 'MESSAGE_TYPE_TEXT'",
+          "instillUIOrder": 3,
+          "instillAcceptFormats": [
+            "string"
+          ],
+          "instillUpstreamTypes": [
+            "value",
+            "reference"
+          ],
+          "title": "Message Type",
           "type": "string"
+        },
+        "duration": {
+          "description": "The duration between now and how long ago to retrieve the chat history from. i.e. 2h45m5s. Valid time units are \"ns\", \"us\" (or \"µs\"), \"ms\", \"s\", \"m\", \"h\".  The default is all with blank setting",
+          "instillUIOrder": 4,
+          "instillAcceptFormats": [
+            "string"
+          ],
+          "instillUpstreamTypes": [
+            "value",
+            "reference"
+          ],
+          "title": "Duration",
+          "type": "string"
+        },
+        "max-message-count": {
+          "description": "The maximum number of messages to retrieve. The default is all with blank setting",
+          "instillUIOrder": 5,
+          "instillAcceptFormats": [
+            "integer"
+          ],
+          "instillUpstreamTypes": [
+            "value",
+            "reference"
+          ],
+          "title": "Max Message Count",
+          "type": "integer"
         }
       },
       "required": [
         "namespace",
         "catalog-id",
-        "file-uid"
+        "conversation-id"
       ],
       "instillEditOnNodeFields": [
         "namespace",
         "catalog-id",
-        "file-uid"
+        "conversation-id"
       ],
       "title": "Input",
       "type": "object"
     },
     "output": {
-      "instillUIOrder": 1,
+      "description": "Chat history from the conversation in catalog",
       "properties": {
-        "succeeded": {
-          "description": "The status of the file processing, if succeeded, return true",
+        "messages": {
+          "description": "Chat history from the conversation in catalog",
           "instillUIOrder": 0,
-          "instillFormat": "boolean",
-          "title": "Status",
-          "type": "boolean"
+          "items": {
+            "properties": {
+              "message-uid": {
+                "description": "The unique identifier of the message",
+                "instillFormat": "string",
+                "instillUIOrder": 0,
+                "title": "Message ID",
+                "type": "string"
+              },
+              "catalog-id": {
```

Comment:
I think all messages in the array should share the same `catalog-id` and `conversation-id`, also matching the user's input. There's no need to return them in the output again.